### ISSUE TITLE
a11y: Desktop switch account menu announced incorrectly/confusingly by screen readers

### ISSUE THREAD

**Issue by patrickhlauke** [bitwarden/clients#5525]
### Steps To Reproduce

1. Start a screen reader (e.g. NVDA on Windows)
2. Go to the desktop application, lock all vaults
3. Navigate the "Switch account" menu using the screen reader

After unlocking at least one of the accounts

4. Navigate to the account menu again

### Expected Result

When all accounts are locked, the "Switch account" menu shows buttons for each account. The account buttons visually have the email address associated with the account, the "Locked" text, and a locked icon. Each button should be announced as "[email address], locked. Switch to this account" or similar.

Once logged in, the menu button visually shows the currently logged-in/used account. Opening the menu, you see any other accounts that you also have access to, and an "Add account button". The menu button itself should be announced as "Logged in as [email address for the currently used account]. Switch account". Any of the other accounts in the menu should be announced as "[email address], locked. Switch to this account" or similar.

### Actual Result

The "Switch account" menu is announced correctly as "Switch account, menu button, collapsed, subMenu". When the menu is opened, it weirdly announced (incorrectly) "dialog, Switch account button, **Logged in as [email address of the first account] on bitwarden.com**". Then, the account buttons, despite showing visually an email address and "locked", are just announced as "button, Switch account".

When already in an unlocked vault, the switch account button is just announced as "[email address of currently logged-in/used account], menu button, collapsed, subMenu" - no indication is given about the purpose of the button (to switch accounts). When opened, it incorrectly announces "dialog, Switch account button, **Logged in as [email address of the first additional account that you can switch to] on bitwarden.com**". Then, the additional account buttons again are just unlabelled "button, Switch account".

### Screenshots or Videos

Video recording on Windows using NVDA of the two above scenarios

https://github.com/bitwarden/clients/assets/895831/34ef860e-0ab4-4851-93fe-db30d98ad37d

### Additional Context

In short, the whole way this menu button, and its options, is announced is currently quite borked, making it difficult or impossible for a screen reader user to actually use it correctly.

### Operating System

Windows

### Operating System Version

11

### Installation method

Direct Download (from bitwarden.com)

### Build Version

Version 2023.4.0 Shell 21.3.1 Renderer 106.0.5249.181 Node 16.16.0 Architecture x64

### Issue Tracking Info

- [X] I understand that work is tracked outside of Github. A PR will be linked to this issue should one be opened to address it, but Bitwarden doesn't use fields like "assigned", "milestone", or "project" to track progress.

**Comment by patrickhlauke** [comment]
![Screenshot showing the button in the open menu with "hello@inclusivedesign24.org / Locked"](https://github.com/bitwarden/clients/assets/895831/5c21db70-e81f-45f4-9f2d-983144f7965f)

```html
<button class="account ng-tns-c178-0 ng-star-inserted" aria-label="Switch account" title="Logged in as hello@inclusivedesign24.org on bitwarden.com.">
  <app-avatar aria-hidden="true" class="ng-tns-c178-0 ng-star-inserted">…</app-avatar><!---->
  <div class="accountInfo ng-tns-c178-0">
    <span aria-hidden="true" class="email ng-tns-c178-0">hello@inclusivedesign24.org</span>
    <!---->
    <span aria-hidden="true" class="status ng-tns-c178-0">Locked</span>
  </div>
  <i aria-hidden="true" class="bwi bwi-2x text-muted ng-tns-c178-0 bwi-lock"></i>
</button>
```

Note the completely incorrect `title` on that button. Also, it shouldn't really be a `title` (which is treated as optional information, depending on how a user navigates...in my video examples above, navigating with cursor keys/reading mode the `title` isn't announced at all), but the actual accessible name itself (i.e. the `aria-label`).

I'll be submitting a PR for this, but...it's overall *very* borked and confused.
